orderbook/src/database/solver_orders.rs

```diff
@@ -0,0 +1,110 @@
+use super::*;
+use crate::integer_conversions::*;
+use anyhow::{anyhow, Context, Result};
+use bigdecimal::BigDecimal;
+use futures::{stream::TryStreamExt, Stream};
+use model::{
+    order::{OrderCreation, OrderKind, SolverOrder},
+    Signature,
+};
+
+impl Database {
+    pub fn solver_orders(&self, min_valid_to: u32) -> impl Stream<Item = Result<SolverOrder>> + '_ {
+        const QUERY: &str = "\
+        SELECT * FROM ( \
```

Comment:
I think it's a shame to have this query logic duplicated only for shortcutting a few where filters (which I assume the query optimizer cuts out anyways).

Reply:
I thought it was an improvement also considering that the frontend facing query might become more complicated in the future or diverge more in general. It is true that logic is duplicated but that also makes this code path simpler.

Reply:
One thing that is missing in this query now is the soft cancellation check. I think this is an example of where it might be easy to forget to update one callsite.

Reply:
I think this should be ok now since the offline cancellation is represented as part of the `invalidated` field now.